### PR TITLE
HIVE-26759: Update SHOW_COMPACTION_QUERY to support postgres HMS

### PR DESCRIPTION
**Pull request by akshat0395**
### What changes were proposed in this pull request?

Update SHOW_COMPACTION_QUERY to support postgres syntax 

### Why are the changes needed?

Details: [HIVE-26759](https://issues.apache.org/jira/browse/HIVE-26759)
`ERROR: column "CC_START" does not exist`, when Postgres is used as Hive metastore

Due to this error, the Initiator class was failing after 1st iteration

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Manually configuring Postgres, Derby, and mysql as HMS and see this error persist

### CODE REVIEW

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java [hunk 1]
```diff
@@ -18,36 +18,34 @@
...
     "SELECT " +
-    "  \"CC_DATABASE\", \"CC_TABLE\", \"CC_PARTITION\", \"CC_STATE\", \"CC_TYPE\", \"CC_WORKER_ID\", " +
-    "  \"CC_START\", \"CC_END\", \"CC_RUN_AS\", \"CC_HADOOP_JOB_ID\", \"CC_ID\", \"CC_ERROR_MESSAGE\", " +
-    "  \"CC_ENQUEUE_TIME\", \"CC_WORKER_VERSION\", \"CC_INITIATOR_ID\", \"CC_INITIATOR_VERSION\", " +
-    "   -1 , \"CC_POOL_NAME\", \"CC_TXN_ID\", \"CC_NEXT_TXN_ID\", \"CC_COMMIT_TIME\", " +
-    "  \"CC_HIGHEST_WRITE_ID\"" +
+    " \"CC_DATABASE\", \"CC_TABLE\", \"CC_PARTITION\", \"CC_STATE\", \"CC_TYPE\", \"CC_WORKER_ID\", " +
+    " \"CC_START\", \"CC_END\", \"CC_RUN_AS\", \"CC_HADOOP_JOB_ID\", \"CC_ID\", \"CC_ERROR_MESSAGE\", " +
+    " \"CC_ENQUEUE_TIME\", \"CC_WORKER_VERSION\", \"CC_INITIATOR_ID\", \"CC_INITIATOR_VERSION\", " +
+    "  -1 AS \"CC_NEXT_TXN_ID\", \"CC_POOL_NAME\", \"CC_TXN_ID\", \"CC_NEXT_TXN_ID\", \"CC_COMMIT_TIME\", " +
```
Comment (deniskuzZ) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java:47]: there are 2 CC_NEXT_TXN_ID declarations

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java [hunk 2]
```diff
@@ -18,36 +18,34 @@
...
 
 public class TxnQueries {
-  public static final String SHOW_COMPACTION_ORDERBY_CLAUSE = 
-    " ORDER BY CASE " +
-    "   WHEN \"CC_END\" > \"CC_START\" and \"CC_END\" > \"CC_COMMIT_TIME\" " +
-    "     THEN \"CC_END\" " +
-    "   WHEN \"CC_START\" > \"CC_COMMIT_TIME\" " +
-    "     THEN \"CC_START\" " +
-    "   ELSE \"CC_COMMIT_TIME\" " +
+  public static final String SHOW_COMPACTION_ORDERBY_CLAUSE = "  ORDER BY  CASE " +
```
[the remark this answers is not in the document]
Comment (deniskuzZ) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java:21]: @akshat0395, thank you for submitting the bug fix. Could you please restore the original formatting, it's very inconvenient to follow the changes.

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java [hunk 3]
```diff
@@ -18,36 +18,34 @@
...
+  public static final String SHOW_COMPACTION_QUERY = "SELECT XX.* FROM ( SELECT " +
+    " \"CQ_DATABASE\" AS \"CC_DATABASE\", \"CQ_TABLE\" AS \"CC_TABLE\", \"CQ_PARTITION\" AS \"CC_PARTITION\", " +
+    " \"CQ_STATE\" AS \"CC_STATE\", \"CQ_TYPE\" AS \"CC_TYPE\", \"CQ_WORKER_ID\" AS \"CC_WORKER_ID\", " +
+    " \"CQ_START\" AS \"CC_START\", -1 AS \"CC_END\", \"CQ_RUN_AS\" AS \"CC_RUN_AS\", " +
+    " \"CQ_HADOOP_JOB_ID\" AS \"CC_HADOOP_JOB_ID\", \"CQ_ID\" AS \"CC_ID\", \"CQ_ERROR_MESSAGE\" AS \"CC_ERROR_MESSAGE\", " +
+    " \"CQ_ENQUEUE_TIME\" AS \"CC_ENQUEUE_TIME\", \"CQ_WORKER_VERSION\" AS \"CC_WORKER_VERSION\", " +
+    " \"CQ_INITIATOR_ID\" AS \"CC_INITIATOR_ID\", \"CQ_INITIATOR_VERSION\" AS \"CC_INITIATOR_VERSION\", " +
+    " \"CQ_CLEANER_START\" AS \"CC_CLEANER_START\", \"CQ_POOL_NAME\" AS \"CC_POOL_NAME\", \"CQ_TXN_ID\" AS \"CC_TXN_ID\", " +
+    " \"CQ_NEXT_TXN_ID\" AS \"CC_NEXT_TXN_ID\", \"CQ_COMMIT_TIME\" AS \"CC_COMMIT_TIME\", " +
+    " \"CQ_HIGHEST_WRITE_ID\" AS \"CC_HIGHEST_WRITE_ID\" " +
```
Comment (InvisibleProgrammer) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java:39]: I agree with @deniskuzZ . Even with ignoring whitespace changes, it is impossible to tell what is the difference.
image

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java (hunk 1 above)
Comment (akshat0395) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java]: Resolved

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java (hunk 3 above)
Comment (akshat0395) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnQueries.java]: Resolved

**Comment by akshat0395** [comment]
@deniskuzZ @InvisibleProgrammer Thanks for the review, I have incorporated the suggestion, request your re-review

**Review by InvisibleProgrammer — APPROVED** [review]

**Review by deniskuzZ — APPROVED** [review]
LGTM, pending tests

**Pull request merged**